docs/guide/adding-custom-widget-placeholder-content.md

```diff
@@ -0,0 +1,126 @@
+# Adding placeholders to custom widgets
+
+Custom widgets can also take advantage of non-schema field placeholders by adding code to the main module, preparing the widget template to show the custom placeholder content, and adding the placeholder content into an accessible folder.
+
+## Basic placeholder addition
+
+### Altering the `index.js` file
+
+ In order to add a custom image file, two options should be set within the main widget file, `placeholder: true` and the `placeholderImage` option with the value set to the file extension of the placeholder content without any `.` prefix.
+
+Next, to supply the URL of the placeholder content, the module should invoke the `determineBestAssetUrl('placeholder')` method in the `init(self)` initialization function.
+
+Even though the option name ends with `Image`, this same method can be used for adding custom video placeholder content that will be uploaded into the module's `public` folder. In contrast, for videos that are hosted externally, you need to add the `placeholder: true` and `placeholderUrl` option with the value set to the URL of the resource. The initialization function in this case can be skipped.
+
+Example for adding a custom image.
+
+<AposCodeBlock>
+
+```javascript
+module.exports = {
+  extend: '@apostrophecms/widget-type',
+  options: {
+    label: 'Custom Widget',
+    placeholder: true,
+    placeholderImage: 'png'
+  },
+  init(self) {
+    self.determineBestAssetUrl('placeholder');
+  },
+  // remainder of the widget code
+};
+```
+
+<template v-slot:caption>
+  modules/custom-widget/index.js
+</template>
+
+</AposCodeBlock>
+
+::: note
+You can bypass the call to `determineBestAsetUrl()` for your images or videos stored in the `public` folder and instead pass the path to the asset using `placeholderUrl` in place of `placeholderImage`. This is discouraged because errors can be made in this path. It is better to let Apostrophe figure this out for you.
+:::
+
+### Altering the widget.html file
+
+The specific alteration of the `widget.html` template will depend on the type of placeholder content being delivered. Irrespective of file type, the template will have access to `data.widget.aposPlaceholder` and `data.manager.options.placeholderUrl` to populate the markup. 
```

Comment:
"But I never set `options.placeholderUrl`!" It is populated for you when you set `placeholderImage` and call `determineBestAssetUrl`.

Reply:
Changed the third paragraph of the section to better clarify.